iFixit/selenium_setup
# Bump Selenium Grid Version

## Description

This change upgrades the Selenium Grid from approximately version 3.14 to around 4.16.

To implement this upgrade, a TOML configuration file is required. This file is essential for specifying the browsers available on the grid and their corresponding drivers. This setup us to ensure that when there's a request for a new session, the `Distributor` component can accurately match the request with browsers that match the "`browserName`" and "`browserVersion`". The upgrade modified the way Selenium pairs browsers, moving away from matching solely based on "`browserName`". Now, the pairing process also considers "`browserVersion`" and "`platformName`".

Reference: https://ifixit.slack.com/archives/C01DBJK0H7U/p1705535705285359

### CR

There will be a follow-up in the `iFixit/ifixit` repo to drop the requirement of a browser version when sending in a session request. Our PHP tests do not cover different browser versions, so this request option is no longer needed. 

### QA

- [ ] Confirm you can run tests with the new local Selenium Grid.

1. Follow the instructions in https://overlord.ifixit.com/Guide/How+to+setup+Selenium+locally/786
2. Check out this branch on Cominor - https://github.com/iFixit/ifixit/pull/51703
3. Run a test on Cominor
	- Example: `platformName=MAC test-suite -sm GuideSmokeTest`


Closes: https://github.com/iFixit/ifixit/issues/51629
